### PR TITLE
F v3 5

### PR DESCRIPTION
**Pull request by toliver**
Tested with v3.6.0.30507

### CODE REVIEW

File: sr_ur_robot_hw/src/sr_ur_robot_state_client.cpp [hunk 1]
```diff
@@ -126,7 +152,7 @@ static void robot_state_received_cb(uv_stream_t* state_stream,
     }
   }
 
-  ur_robot_state *robot_state = reinterpret_cast<ur_robot_state*>(pdata);
+  ur_robot_state_v3_5 *robot_state = reinterpret_cast<ur_robot_state_v3_5*>(pdata);
```
Comment (beatrizleon) [inline sr_ur_robot_hw/src/sr_ur_robot_state_client.cpp:155]: @toliver do we need an if here to have either ur_robot_state_v3_2 or ur_robot_state_v3_5?

File: sr_ur_robot_hw/src/sr_ur_robot_state_client.cpp [hunk 2]
```diff
@@ -169,8 +195,9 @@ static void robot_state_client_connected_cb(uv_connect_t* connection_request, in
   ROS_ASSERT(0 == status);
   ROS_ASSERT(connection_request == &rs_client->connection_request_);
 
-  rs_client->buffer_.base = reinterpret_cast<char*>(malloc(2*sizeof(ur_robot_state)));
-  rs_client->buffer_.len  = 2*sizeof(ur_robot_state);
+  // Use the longest structure to allocate memory (ur_robot_state_v3_5)
+  rs_client->buffer_.base = reinterpret_cast<char*>(malloc(2*sizeof(ur_robot_state_v3_5)));
+  rs_client->buffer_.len  = 2*sizeof(ur_robot_state_v3_5);
```
Comment (beatrizleon) [inline sr_ur_robot_hw/src/sr_ur_robot_state_client.cpp:200]: @toliver same as above?

File: sr_ur_robot_hw/src/sr_ur_robot_state_client.cpp (hunk 1 above)
Comment (toliver) [reply sr_ur_robot_hw/src/sr_ur_robot_state_client.cpp:155]: Not really, as we never access the last fields of these structures.
That's also the reason why we use the size of the longest structure to allocate memory for the buffer.

File: sr_ur_robot_hw/src/sr_ur_robot_state_client.cpp (hunk 2 above)
Comment (toliver) [reply sr_ur_robot_hw/src/sr_ur_robot_state_client.cpp:200]: Not really (it is explained in the comment actually).

// Use the longest structure to allocate memory (ur_robot_state_v3_5)

File: sr_ur_robot_hw/src/sr_ur_robot_state_client.cpp (hunk 1 above)
Comment (toliver) [reply sr_ur_robot_hw/src/sr_ur_robot_state_client.cpp:155]: I agree this should be refactored to have a nicer way of coping with the different structures.
Actually we might be OK with simply assuming everything is cast to a structure with the latest version.
The reason is that for the moment all the data fields we care about are common to all versions. The new data fields that are added in each subsequent version are not used in this driver.
Although maybe at some point we might want to use them or publish them, so maybe we should distinguish versions.

**Pull request merged**